component/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio::sync::oneshot::{
+    channel as oneshotChannel, Receiver as oneshotReceiver, Sender as oneshotSender,
+};
+
+pub type IrrecoverableError = anyhow::Error;
+type JoinHandle = tokio::task::JoinHandle<()>;
```

Comment:
Should we force the join handle to be of `JoinHandle<Result<(), Error>>` to allow the task to return with an error, and then pass in this error to restart. Of course if it ends without an error we should allow the `Manageable` task to exit.

Reply:
My understanding of our intention for these components is for them to be constantly up and running, and if it were possible to somehow end the component, it would constitute as a Byzantine failure. If that is still true, then sending an error in addition to the one passed in the channel may be a bit redundant.